AIFS_ROS/hiperlab_rostools/src/AirSimBridge/main.cpp

```diff
@@ -77,24 +85,104 @@ class ImagePollAgent {
       cv::Mat depthImage;
       cv::Mat depthImage_uint8;
 
-//        {
-////            This part of the code saves image to local. Uncomment for debug purpose.
-//                      char *homdir = getenv("HOME");
-//                      string homdirstr = homdir;
-//                      path = homdirstr + "/Documents/AirSim";
-//                      char buffer[256];
-//                      sprintf(buffer, "%04d", imageCount - 3);
-//                      std::string str(buffer);
-//                      std::string file_path = common_utils::FileSystem::combine(path, "img" + str);
-//                      std::ofstream file(file_path + ".png", std::ios::binary);
-//                      file.write(
-//                          reinterpret_cast<const char*>(image_info.image_data_uint8.data()),
-//                          image_info.image_data_uint8.size());
-//                      file.close();
-//        }
+      /*{
+       //            This part of the code saves image to local. Uncomment for debug purpose.
+       char *homdir = getenv("HOME");
+       string homdirstr = homdir;
+       path = homdirstr + "/Documents/AirSim";
+       char buffer[256];
+       sprintf(buffer, "%04d", imageCount - 3);
+       std::string str(buffer);
+       std::string file_path = common_utils::FileSystem::combine(path,
+       "img" + str);
+       std::ofstream file(file_path + ".png", std::ios::binary);
+       file.write(
+       reinterpret_cast<const char*>(response.at(0).image_data_uint8.data()),  //image_info.image_data_uint8.data()),
+       image_info.image_data_uint8.size());
+       file.close();
+       } */
+
+      depthImage_uint8 = cv::imdecode(response.at(0).image_data_uint8,
+                                      cv::IMREAD_UNCHANGED);
+
+      ros::Time timeStartEncode = ros::Time::now();
+      ros::Duration fetchImageTime = timeStartEncode - timeStartFetch;
+      std_msgs::Header header;  // empty header
+      header.seq = imageCount - 2;  // user defined counter
+      header.stamp = ros::Time::now();  // time
+      cv_bridge::CvImage img_bridge;
+      sensor_msgs::Image img_msg;  // >> message to be sent
+
+      img_bridge = cv_bridge::CvImage(header,
+                                      sensor_msgs::image_encodings::RGB8,
+                                      depthImage_uint8);
+
+      img_bridge.toImageMsg(img_msg);  // from cv_bridge to sensor_msgs::Image
+      pubDepthImage->publish(img_msg);  // ros::Publisher pub_img = node.advertise<sensor_msgs::Image>("topic", queuesize);
+      ros::Time timePublish = ros::Time::now();
+
+      std_msgs::Header imageReceived;  // empty header
+      imageReceived.seq = imageCount;  // user defined counter
```

Comment:
Sorry about my legacy code. Maybe wrap the performance evaluation part within an if statement so we can toggle if we are only running it when we want to debug.